index.js

```diff
@@ -20,6 +22,106 @@ app.get('/info', (req, res) => {
   res.send({ serverName: packageData.name, serverVersion: packageData.version });
 });
 
+// route for displaying all products
+app.get('/products/all', (req, res) => {
+
+  var jsonOutputObject = {
+    product_list_categorically : []
+  };
+
+  for(var i = 0; i < categoryData.categories.length; i++) {
+    
+    // JSON to be inserted in the main object
+    var jsonObjectToBeInserted = {
+      categoryId : categoryData.categories[i].id,
+      categoryName : categoryData.categories[i].categoryName,
+      products : []
+    }
+
+    for(var j = 0; j < productData.products.length; j++) {
+      if (productData.products[j].categoryId == categoryData.categories[i].id) {
+        jsonObjectToBeInserted.products.push(productData.products[j]);
+      }
+    }
+
+    jsonOutputObject.product_list_categorically.push(jsonObjectToBeInserted);
+  }
+
+  res.send(jsonOutputObject);
+});
+
+// Getting product info from its ID
+app.get('/product/:id', (req, res) => {
+
+  let userInput = req.params.id;
+  var productId;
+  var jsonOutputObject = null;
+
+  // Generating valid ID and returning error message when ID is invalid
+  if (userInput < 1) {
+    res.send('Invalid product ID');
+    return;
+  } else if (userInput >= 1 && userInput <= 9) {
```

Comment:
What do you think about the following code snippet to avoid multiple if condtions
```
productId = `P${userInput.padStart(3, '0')}`
```

Reply:
Yeah, it's better to templates and padding which is faster and reduces code. Thanks for suggestation :)